br/pkg/lightning/mydump/region.go

```diff
@@ -359,7 +359,7 @@ func SplitLargeFile(
 		if err != nil {
 			return 0, nil, nil, err
 		}
-		parser, err := NewCSVParser(&cfg.Mydumper.CSV, r, int64(cfg.Mydumper.ReadBlockSize), ioWorker, true, nil)
```

Comment:
wouldn't actually constructing a `NewCharsetConvertor` replacing the `nil` here also fix the problem? and the diff would be much smaller.

Reply:
My concern is that this way every time you build a CSV parser you need to consider whether to pass a charset convertor and if not, the charset setting may not work globally, which is quite like an undefined behavior. Putting its construction inside the `NewCSVParser` seems safer.

Reply:
it should always be passed? the only cases where you provide `nil` unconditionally is in tests.

Reply:
Yes, you're right. So is passing it rather than making it inside more reasonable🤔？

Reply:
i just find the diff to be bigger than necessary 😅